src/Components/CommentCard.jsx

```diff
@@ -1,29 +1,50 @@
-import {useState} from 'react';
+import {useEffect, useState} from 'react';
+import { useContext } from 'react';
+import { UserNameContext } from '../Contexts/Users';
+import { deleteComment } from '../../api';
+import { getCommentsByArticle } from '../../api';
 
-const CommentCard = ({prop}) => {
+const CommentCard = ({prop, states}) => {
     const [votes, setVotes] = useState(prop.votes);
     const [isDisabled, setDisable] = useState(false)
     const incrementVote = () => {
         setVotes(votes+1)
         setDisable(true)
-
     }
+    const {userName, setUserName} = useContext(UserNameContext);
+    const [commentDisplay, setCommentDisplay] = useState(prop)
+  
+    
+    
+    
     const decrementVote = () => {
         if (votes>0)
         setVotes(votes-1)
     }
-    
-  
+    const handleDeleteBTN = (event) => {
+
+        event.preventDefault()
+        deleteComment(prop.comment_id).then(()=>{
+            return getCommentsByArticle(prop.article_id)
+        }).then(({comments}) => {
+            setCommentDisplay(comments)
+        });
+        window.alert(`${prop.author} has deleted their comment`)
+    }
+        
```

Comment:
It would probably be good from a user experience standpoint to have the alert pop up after the request to delete from the API resolves in the `then` , otherwise they will get the confirmation even if it fails to delete 😬